pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -236,38 +248,27 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	app.Status.Phase = common.ApplicationRunning
 
 	if err := garbageCollection(ctx, handler); err != nil {
-		klog.ErrorS(err, "Failed to run Garbage collection")
+		klog.ErrorS(err, "Failed to run garbage collection")
 		r.Recorder.Event(app, event.Warning(velatypes.ReasonFailedGC, err))
 		return handler.handleErr(err)
 	}
 	klog.Info("Successfully garbage collect", "application", klog.KObj(app))
 
-	// Gather status of components
-	var refComps []v1alpha1.TypedReference
-	for _, comp := range comps {
-		refComps = append(refComps, v1alpha1.TypedReference{
-			APIVersion: comp.APIVersion,
-			Kind:       comp.Kind,
-			Name:       comp.Name,
-			UID:        app.UID,
-		})
-	}
-	app.Status.Components = refComps
```

Comment:
will there be any compatible issue? Considing existing Application to do update/delete.

Reply:
Since we already replace AppCtx with assemble&dispatch,  this will not make compatible issues. This `Status.Components` filed is not used except here. And components are deleted by gc through having app as their owners.